### PR TITLE
Update README introduction

### PR DESCRIPTION
**Pull request by mxstbr**
Based on the discussion in #436, a first draft of an update to the introduction text

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,24 +2,13 @@
 
 [![Join the chat at https://gitter.im/yelouafi/redux-saga](https://badges.gitter.im/yelouafi/redux-saga.svg)](https://gitter.im/yelouafi/redux-saga?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge) [![npm version](https://img.shields.io/npm/v/redux-saga.svg?style=flat-square)](https://www.npmjs.com/package/redux-saga)
 
-An alternative Side Effects middleware (aka Asynchronous Actions) for Redux applications. Instead of dispatching Thunks which get handled by the `redux-thunk` middleware, you create *Sagas* to gather all your Side Effects logic in a central place.
+`redux-saga` is a library that aims to make asynchronous flows (e.g. data fetching) in React.js applications easier and better.
```
Comment (yelouafi) [inline README.md]: Do you think we can borrow something from this http://survivejs.com/blog/redux-saga-interview/#redux-middlewares. Mainly, define what 'Side Effects' are in the context of Redux (async and impure calls).

File: README.md [hunk 2]
```diff
@@ -2,24 +2,13 @@
...
 
-This means application logic lives in 2 places:
+The mental model is that a saga is like a separate thread in your application that's solely responsible for asynchronousity. This thread can be started, paused and cancelled from the main application with normal redux actions, it has access to the full redux application state and it can dispatch redux actions as well.
```
Comment (yelouafi) [inline README.md]: While this is correct. There are also things that are synchronous but not pure (like saving to a browser storage, clearing a timeout ID, ...). If we define Side Effects in the first paragraph, we can simply put here that a Saga is solely responsible for Side Effects, or to reflect the opinions of the library: redux-saga proposes to 'isolate' all sorts of Side Effects (including async calls) inside Sagas.

File: README.md [hunk 3]
```diff
@@ -2,24 +2,13 @@
...
 
-- Reducers are responsible for handling state transitions between actions.
-- Sagas are responsible for orchestrating complex/asynchronous operations.
+It uses a soon-to-be-built-in JavaScript feature called Generators to make those asynchronous flows easy to read, write and test. *(if you're not familiar with [here are some introductory links](https://yelouafi.github.io/redux-saga/docs/ExternalResources.html))* By doing so, these asynchronous flows look like your standard synchronous JavaScript code. (kind of like `async`/`await`, but generators have a few awesome features we need)
```
Comment (yelouafi) [inline README.md]: Perhaps worth mentioning that Generators are part of the ES6 standard

**Comment by yelouafi** [comment]
sorry for the delay. I added some notes.

**Comment by mxstbr** [comment]
Updated with your feedback! Let me know what you think, I definitely think this is a lot clearer.

**Comment by yelouafi** [comment]
Looks great! thanks.

Sorry but one last remark, actually there is no mention that this a Redux middleware. Perhaps we should mention that

- The library is a redux middleware (although it can be used outside of Redux)  
-  [Optionally define what is Redux] a library which makes managing application state in consistent and predictable
- redux without middleware can not perform side effects but only transform state inside reducers

**Comment by mxstbr** [comment]
> The library is a redux middleware (although it can be used outside of Redux)

:+1: Done

> [Optionally define what is Redux] a library which makes managing application state consistent and predictable
> redux without middleware can not perform side effects but only transform state inside reducers

I don't think we should add this to the introductory text, I think assuming that people know redux is a fair assumption for a library called `redux-saga` that mentions it's a redux middleware

**Comment by yelouafi** [comment]
>I don't think we should add this to the introductory text, I think assuming that people know redux is a fair assumption for a library called redux-saga that mentions it's a redux middleware

Right.

I'll update the docs repo right away. Thanks again!

**Comment by mxstbr** [comment]
Awesome! 🎉

**Pull request merged**